discv5/discv5-theory.md

```diff
@@ -41,6 +40,141 @@ IP address and port.
 If the endpoint cannot be determined (e.g. when the NAT doesn't support 'full-cone'
 translation), implementations should omit IP address and UDP port from the record.
 
+## Sessions
+
+Discovery communication is encrypted and authenticated using session keys, established in
+the handshake.
+
+### Handshake
+
+Since every node participating in the network acts as both client and server, a handshake
+can be initiated by either side of communication at any time. In the following
+definitions, we assume that node A wishes to communicate with node B, e.g. to send a
+FINDNODE query.
+
+Node A must have a node record for node B and know B's node ID to communicate with it. If
+node A has session keys from prior communication, it encrypts its request with those keys.
+If no keys are known, it initiates the handshake by sending a packet with random content.
```

Comment:
Is there any minimal random message size requirement as in v5.0?

Reply:
I'm still working on it.

Reply:
This is added now.